src/main.js

```diff
@@ -27,27 +106,78 @@ const findInstalledBrowsers = () => {
       reject(data)
     })
     sp.stdout.on('end', () => {
-      profile = parser.toJson(profile, { object: true })
+      profile = parser.toJson(profile, {
+        object: true
+      })
       const installedApps = jp.query(
         profile,
         'plist.array.dict.array[1].dict[*].string[0]'
       )
-      const installedBrowsers = installedApps
-        .map(appName => {
-          for (let i = 0; i < browsers.length; i++) {
-            const browser = browsers[i]
-            if (browser.name === appName) {
-              return browser
+      //TODO: Delete old code
+
+      // const installedBrowsers = installedApps
+      //   .map(appName => {
+      //     for (let i = 0; i < browsers.length; i++) {
+      //       const browser = browsers[i]
+      //       if (browser.name === appName) {
+      //         return browser
+      //       }
+      //     }
+      //     return false
+      //   })
+      //   .filter(x => x) // remove empties
+      //
+
+      // NOTE: Algorithmically speaking this whole thing is an overkill, but working with small numers it will be fine
+      const installedBrowsers = configUser.browsers
+        .map(browser => {
+          // Quoting @will-stone from https://github.com/will-stone/browserosaurus/issues/13
+          // Not on defaults list, not in profiler results: *Notification says: "Google Chrome Error" not currently supported or found on this Mac.
+
+          if (installedApps.indexOf(browser.name) == -1) {
+            notifications.push(
+              new Notification('error', 'Browser/app not found.')
+            )
+            return false
+          } else {
+            // Not on defaults list, in profiler results (therefore shown with no icon): *Notification says: "Beaker Browser" not officially supported, please ask Browserosaurus to add this browser.
+            if (
+              configDefault.browsers
+                .map(defaultBrowser => {
+                  if (defaultBrowser.name == browser.name) {
+                    return true
+                  } else {
+                    return false
+                  }
+                })
+                .filter(x => x).length == 0
+            ) {
+              notifications.push(
+                new Notification(
+                  'warning',
+                  'Youtra dev, I knew it! Please open an issue about this wonderful browser..'
+                )
+              )
+              browser.icon = 'Custom'
             }
+            return browser
           }
-          return false
         })
-        .filter(x => x) // remove empties
+        .filter(x => x)
+
+      // TODO: remove debugging printouts
+      console.log(notifications)
+
       fulfill(installedBrowsers)
     })
   })
 }
 
+// TODO: Something, but not now
+function mergeConfigs(localConfig, userConfig) {
+  return userConfig
```

Comment:
I don’t think this is doing anything, but correct me if I’m wrong. As I previously mentioned, I really don’t think merge is required and will potentially break future GUI settings mode. Until then, you know how it works so updating your own config will be easy. Let’s leave it out please.